### PR TITLE
DDT-1295 added TO's related to shipping instructions

### PR DESCRIPTION
**Pull request by pimg**
resolves: DDT-1295

### CODE REVIEW

File: edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/EquipmentTO.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  @Size(max = 15)
+  String equipmentReference,
+
+  @JsonProperty("ISOEquipmentCode")
+  @Size(max = 4)
+  String isoEquipmentCode,
+
+  Double tareWeight,
+
+  WeightUnit weightUnit
```
Comment (nt-as) [inline edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/EquipmentTO.java:21]: Should we have a `@RequiredIfOther(ifNotNull = "tareWeight", thenNotNull = "weightUnit")` here?

File: edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/UtilizedTransportEquipmentValidator.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    if (utilizedTransportEquipmentTO != null) {
+      if (utilizedTransportEquipmentTO.isShipperOwned()) {
+        return containsWeightAndUnit(utilizedTransportEquipmentTO.equipment());
+      }
+      return true;
+    }
+    return false;
+  }
+
+  private Boolean containsWeightAndUnit(EquipmentTO equipment) {
```
Comment (nt-as) [inline edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/UtilizedTransportEquipmentValidator.java:31]:
```suggestion
  private boolean containsWeightAndUnit(EquipmentTO equipment) {
```

It cannot return `null`.

File: edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/ValidUtilizedTransportEquipment.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Validate that a given String field contains a valid vessel IMO number
```
Comment (nt-as) [inline edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/ValidUtilizedTransportEquipment.java:14]: Looks like copy-waste given it does not validate IMO numbers.

File: edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/EquipmentTO.java (hunk 1 above)
Comment (pimg) [reply edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/EquipmentTO.java:23]: fixed

File: edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/UtilizedTransportEquipmentValidator.java (hunk 2 above)
Comment (pimg) [reply edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/UtilizedTransportEquipmentValidator.java]: fixed

File: edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/ValidUtilizedTransportEquipment.java (hunk 3 above)
Comment (pimg) [reply edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/ValidUtilizedTransportEquipment.java]: fixed

**Review by nt-as — APPROVED** [review]

**Pull request merged**